Assets/Scripts/Input/DragonPlayerController.cs

```diff
@@ -162,12 +162,14 @@ protected override void HandleDeathEvent()
     {
         base.HandleDeathEvent();
         movement.ToggleGravity(true);
+        PlayerManager.Instance.IncrementDeathCount();
     }
 
-    protected override void HandleReviveFinshEvent()
+    protected override void HandleReviveFinishEvent()
     {
-        base.HandleReviveFinshEvent();
+        base.HandleReviveFinishEvent();
         movement.ToggleGravity(false);
         movement.TakeFlight();
+        PlayerManager.Instance.DecrementDeathCount();
```

Comment:
Maybe create a `HandleReviveStartEvent` in the `ActorController` and add it as a listener to `Combat.ReviveStartEvent`, then place the decrementing of death count there instead. Decrementing only after revival has finished could cause a case where for example the dragon finishes the revive interaction but dies while the knight is getting up, causing the death count to reach 2.